src/Sylius/Component/Product/Model/DateRange.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Product\Model;
+
+/**
+ * @author Krzysztof Wędrowicz <krzysztof@wedrowicz.me>
+ */
+class DateRange
+{
+    /**
+     * @var \DateTime
+     */
+    private $start;
+
+    /**
+     * @var \DateTime
+     */
+    private $end;
+
+    /**
+     * @param \DateTime $start
+     * @param \DateTime $end
+     */
+    public function __construct(\DateTime $start = null,\DateTime $end = null)
+    {
+        $this->start = $start;
+        $this->end = $end;
+    }
+
+    /**
+     * Checks whether current datetime meets two requirements:
+     * is greater than $start or equal,
+     * is smaller than $end or equal.
+     * Checks only these conditions which values ($start, $end) are not null.
+     *
+     * @return bool
+     */
+    public function isInRange()
+    {
+        if($this->start > new \DateTime() || ($this->end !== null && $this->end < new \DateTime())){
```

Comment:
Nope, because null cant be greater than anything so this condition would be redundant :)